### PR TITLE
3 cleanup commits

### PR DESCRIPTION
**Pull request by kevmoo**

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,10 +1,7 @@
 # Don’t commit the following directories created by pub.
-build/
```
Comment (isoos) [inline .gitignore]: please revert, `pkg/code_coverage` emits data in the `build/` directory.

File: .gitignore [hunk 2]
```diff
@@ -1,10 +1,7 @@
...
-packages
-.buildlog
 .packages
-.pub
 .dart_tool
+pkg/**/pubspec.lock
```
Comment (isoos) [inline .gitignore]: Please revert, we've decided to include the pubspec.lock files in all of these projects (caused CI failures to not track the dependencies).

Comment (kevmoo) [reply .gitignore]: There were two things in `/pkg` that didn't have lock files.

Should I include those in this CL or leave it for later?

Comment (isoos) [reply .gitignore]: Up to you, feel free to leave it later.

**Comment by kevmoo** [comment]
@isoos PTAL

**Comment by isoos** [comment]
`lib/search/scoring.dart` needs `dartfmt`

**Comment by kevmoo** [comment]
PTAL @isoos

**Pull request merged**